lib/shelley/src/Cardano/Wallet/Shelley/Transaction.hs

```diff
@@ -1449,6 +1453,28 @@ _assignScriptRedeemers pparams ti resolveInput redeemers tx =
                 }
             }
 
+eraFromPParams :: ProtocolParameters -> AnyCardanoEra
```

Comment:
what about defining this in `Cardano.Wallet.Primitive.Types` where `ProtocolParameters` is defined? Or  `Cardano.Wallet.Primitive.Types.Tx` where AnyCardanoEra is defined (ie. reimported from Cardano.Api)?